### ISSUE TITLE
`implicit_return` false positive inside `tauri::command` macro

### ISSUE THREAD

**Issue by jsimonrichard** [rust-lang/rust-clippy#12872]
### Summary

I'm denying `implicit_return` (for readability) with a Tauri project which includes async commands. Somewhere inside the `tauri::command` macro or the `tauri::generate_handler` macro, there's an implicit return that clippy is picking up on (but only when the command function is async).

But it's my understanding that clippy lints like this one should not be applied to the code inside macros (especially external ones). And obviously I can't control the code style of an external crate.

### Lint Name

implicit_return

### Reproducer

I've provided a minimal reproducible example repo here: https://github.com/jsimonrichard/clippy-implicit_return-tauri-command-macro-issue-mre

I saw this happen:

```
error: missing `return` statement
  --> src/main.rs:5:1
   |
5  | #[tauri::command]
   | ^^^^^^^^^^^^^^^^^ help: add `return` as shown: `return #[tauri::command]`
...
12 |         .invoke_handler(tauri::generate_handler![greet])
   |                         ------------------------------- in this macro invocation
```

I expected to see this happen: no error.

### Version

```text
rustc 1.78.0 (9b00956e5 2024-04-29)
binary: rustc
commit-hash: 9b00956e56009bab2aa15d7bff10916599e3d6d6
commit-date: 2024-04-29
host: x86_64-unknown-linux-gnu
release: 1.78.0
LLVM version: 18.1.2
```
